sparc.patch by Jurij Smakov <jurij@wooyd.org>

(1) Defines ARCH=ultrasparc for any sparc* architecture. This allows
    one to build HiPE enabled package for sparc in Debian.

(2) Fixes FTBFS(see http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=328031).

--- a/erts/configure.in
+++ b/erts/configure.in
@@ -615,7 +615,7 @@
 fi
 
 case $chk_arch_ in
-    sun4u)	ARCH=ultrasparc;;
+    sun4u|sparc*)	ARCH=ultrasparc;;
     sparc64)	ARCH=sparc64;;
     sun4v)	ARCH=ultrasparc;;
     i86pc)	ARCH=x86;;
--- a/erts/emulator/beam/beam_emu.c
+++ b/erts/emulator/beam/beam_emu.c
@@ -1105,19 +1105,11 @@
  * in registers.
  */
 
-#if defined(__GNUC__) && defined(sparc) && !defined(DEBUG)
-#  define REG_xregs asm("%l1")
-#  define REG_htop asm("%l2")
-#  define REG_stop asm("%l3")
-#  define REG_I asm("%l4")
-#  define REG_fcalls asm("%l5")
-#else
 #  define REG_xregs
 #  define REG_htop
 #  define REG_stop
 #  define REG_I
 #  define REG_fcalls
-#endif
 
 #ifdef USE_VM_PROBES
 #  define USE_VM_CALL_PROBES
